test/integration/revocation_integration_test.go

```diff
@@ -0,0 +1,308 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	secretsv1beta1 "github.com/hashicorp/vault-secrets-operator/api/v1beta1"
+	"github.com/hashicorp/vault-secrets-operator/internal/consts"
+)
+
+// TestRevocation tests the revocation logic on Helm uninstall
+func TestRevocation(t *testing.T) {
+	if !testWithHelm {
+		t.Skipf("Test is only compatiable with Helm")
+	}
+	ctx := context.Background()
+	testID := strings.ToLower(random.UniqueId())
+	testK8sNamespace := "k8s-tenant-" + testID
+	testKvv2MountPath := consts.KVSecretTypeV2 + testID
+	testVaultNamespace := ""
+	testNamePrefix := "revocation"
+	k8sConfigContext := os.Getenv("K8S_CLUSTER_CONTEXT")
+	if k8sConfigContext == "" {
+		k8sConfigContext = "kind-" + clusterName
+	}
+	require.NotEmpty(t, clusterName, "KIND_CLUSTER_NAME is not set")
+	operatorNS := os.Getenv("OPERATOR_NAMESPACE")
+	require.NotEmpty(t, operatorNS, "OPERATOR_NAMESPACE is not set")
+
+	// TF related setup
+	tempDir, err := os.MkdirTemp(os.TempDir(), t.Name())
+	require.Nil(t, err)
+	tfDir, err := files.CopyTerraformFolderToDest(
+		path.Join(testRoot, "revocation/terraform"),
+		tempDir,
+		"terraform",
+	)
+	require.Nil(t, err)
+
+	// Construct the terraform options with default retryable errors to handle the most common
+	// retryable errors in terraform testing.
+	tfOptions := &terraform.Options{
+		// Set the path to the Terraform code that will be tested.
+		TerraformDir: tfDir,
+		Vars: map[string]interface{}{
+			"k8s_vault_connection_address": testVaultAddress,
+			"k8s_test_namespace":           testK8sNamespace,
+			"k8s_config_context":           k8sConfigContext,
+			"vault_kvv2_mount_path":        testKvv2MountPath,
+			"operator_helm_chart_path":     chartPath,
+			"name_prefix":                  testNamePrefix,
+		},
+	}
+	if operatorImageRepo != "" {
+		tfOptions.Vars["operator_image_repo"] = operatorImageRepo
+	}
+	if operatorImageTag != "" {
+		tfOptions.Vars["operator_image_tag"] = operatorImageTag
+	}
+	if entTests {
+		testVaultNamespace = "vault-tenant-" + testID
+		tfOptions.Vars["vault_enterprise"] = true
+		tfOptions.Vars["vault_test_namespace"] = testVaultNamespace
+	}
+	tfOptions = setCommonTFOptions(t, tfOptions)
+
+	crdClient := getCRDClient(t)
+	var created []ctrlclient.Object
+
+	// Run "terraform init" and "terraform apply". Fail the test if there are any errors.
+	terraform.InitAndApply(t, tfOptions)
+
+	// Parse terraform output
+	b, err := json.Marshal(terraform.OutputAll(t, tfOptions))
+	require.Nil(t, err)
+
+	var outputs revocationK8sOutputs
+	require.Nil(t, json.Unmarshal(b, &outputs))
+
+	// Set the secrets in vault to be synced to kubernetes
+	vClient := getVaultClient(t, testVaultNamespace)
+
+	auths := []*secretsv1beta1.VaultAuth{
```

Comment:
Okay - now I see why the tf code had the JWT/oidc setup resources in it.
nit: I'm pretty sure that the test is still going to be valid (and doesnt really change?) using only the k8s auth method which can simplify the test a bit? Not going to block on this, but would be nice to trim it up and use only k8s auth.

Reply:
I will trim and simplify the test. Thank you!